nbconvert/preprocessors/execute.py

```diff
@@ -64,13 +64,27 @@ class ExecutePreprocessor(Preprocessor):
 
     extra_arguments = List(Unicode())
 
+    kernel_name = Unicode(
+        "python", config=True,
```

Comment:
Yeah I think retaining current behavior and just making the override an
additional option is the right thing to do.
On Dec 23, 2015 11:37 AM, "Antonino Ingargiola" <notifications@github.com>
wrote:

> In nbconvert/preprocessors/execute.py
> <https://github.com/jupyter/nbconvert/pull/177#discussion_r48373573>:
>
> > @@ -64,13 +64,27 @@ class ExecutePreprocessor(Preprocessor):
> >
> >      extra_arguments = List(Unicode())
> >
> > +    kernel_name = Unicode(
> > +        "python", config=True,
>
> Absolutely, it needs to be changed to None (otherwise it always overrides
> the kernel_spec kernel).
>
> Thinking about defaults, note that even before this PR, the kernel name is
> set to "python" when no name is specified in the kernel_spec (see this
> line
> <https://github.com/jupyter/nbconvert/blob/4.1.0/nbconvert/preprocessors/execute.py#L73>).
> Apparently on both python2 and 3 the is always a kernel called python
> which points to the current python version. This is why I did catch the
> wrong default in the kernel_name traitlet.
>
> In principle no kernel in kernel_spec and no explicit kernel_name passed
> should raise an error. However, I'm a bit reluctant adding this exception
> because it will break backward compatibility if somebody is relying on
> automatic selection of "python" kernel when no kernel is specified in the
> notebook metadata. On the other hand, it will make more explicit which
> kernel the user wants to use.
>
> So, what do you guys prefer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/nbconvert/pull/177/files#r48373573>.
>